models/users.js

```diff
@@ -790,21 +791,17 @@ const getUsersByRole = async (role) => {
   }
 };
 
-const generateUniqueUsername = async (firstname, lastname) => {
+const generateUniqueUsername = async (firstName, lastName) => {
   try {
-    const snapshot = await userModel
-      .where("first_name", "==", firstname)
-      .where("last_name", "==", lastname)
-      .count()
-      .get();
-    const existingUserCount = snapshot.data().count;
-    const initialUsername = `${firstname}-${lastname}`;
-    if (existingUserCount === 0) {
-      return initialUsername;
-    } else {
-      const uniqueUsername = `${initialUsername}-${existingUserCount + 1}`;
-      return uniqueUsername;
-    }
+    const snapshot = await userModel.where("first_name", "==", firstName).where("last_name", "==", lastName).get();
+
+    const existingUserCount = snapshot.size;
```

Comment:
I think there is something in firestore that directly returns count have a look once

Reply:
fixed